charts/victoria-metrics-distributed/README.md

```diff
@@ -0,0 +1,209 @@
+
+### Why use `victoria-metrics-distributed` chart?
+
+One of the best practice of running production kubernetes cluster is running with [multiple availability zones](https://kubernetes.io/docs/setup/best-practices/multiple-zones/). And apart from kubernetes control plane components, we also want to spread our application pods on multiple zones, to continue serving even if zone outage happens.
+
+VictoriaMetrics supports [data replication](https://docs.victoriametrics.com/cluster-victoriametrics/#replication-and-data-safety) natively which can guarantees data availability when part of the vmstorage instances failed. But it doesn't works well if vmstorage instances are spread on multiple availability zones, since data replication could be stored on single availability zone, which will be lost when zone outage happens.    
+To avoid this, vmcluster must be installed on multiple availability zones and each of them should contain complete data. In this way, as long as one zone is available, both global write and read entrypoint work without interruption.
+
+### How to write data?
+
+The chart provides `vmauth-global-write` as global write entrypoint, it supports [push-based data ingestion protocols](https://docs.victoriametrics.com/vmagent/#how-to-push-data-to-vmagent) as VictoriaMetrics does.
+You can use this entrypoint as vmagent's remoteWrite url or use other clients push data to it directly.
+
+### How to query data?
+
+The chart provides `vmauth-global-read` as global read entrypoint, it picks the first zone as it's preferred datasource and switches automatically to next zone if first one is unavailable, check [vmauth `first_available`](https://docs.victoriametrics.com/vmauth/#high-availability) for more details.  
+If you have workloads like vmalert or grafana which deployed on one zone specifically, you can use `vmauth-read-proxy` on the same zone as your datasource, since it always prefer "local" vmcluster which reduces cross-zone traffic. 
+
+You can also pick other proxies like kubernetes service which supports [Topology Aware Routing](https://kubernetes.io/docs/concepts/services-networking/topology-aware-routing/) as global read entrypoint.
+
+### What happens if zone outage happened?
+
+If availability zone `zone-eu-1` had an outage, `vmauth-global-write` and `vmauth-global-read` still work without interruption, and following things happen:
+1. `vmauth-global-write` stops proxying write requests to `zone-eu-1` automatically;
+2. `vmauth-global-read` and `vmauth-read-proxy` stops proxying read requests to `zone-eu-1` automatically;
+3. `vmagent` on `zone-us-1` fails to send data to `zone-eu-1` `vmauth-write-balancer`, starts to [buffer data on disk](https://docs.victoriametrics.com/vmagent/#calculating-disk-space-for-persistence-queue) if `-remoteWrite.disableOnDiskQueue` is not specified.
+
```

Comment:
Once the availability of `zone-eu-1` is restored, `vmagent` from `zone-us-1` will re-send all the buffered data to `zone-eu-1` to restore data consistency.

Reply:
@Haleygo @zekker6 when will vmauth start to route requests to `zone-eu-1`? Could it be that after the outage vmauth will route queries to `zone-eu-1` and will serve incomplete responses?

Reply:
Based on [this](https://github.com/VictoriaMetrics/helm-charts/blob/64b4f71f94139548de4147c06a0907983901aaab/charts/victoria-metrics-distributed/victoriametrics-distributed-topology.png) diagram it will start routing requests after vmauth read proxy will become available. So in this case it is possible that it will deliver incomplete responses.

Reply:
Add more description for it, plz check https://github.com/VictoriaMetrics/helm-charts/pull/960/commits/19f29bedc9a77a5c285cc84bfdca32ef3f8ac80c